ruby/vampires.rb

```diff
@@ -43,11 +64,31 @@
 	obvious_name = false
 end
 
+if allergies != "sunshine"
 while obvious_name == false
 	if match == true && (willing == true || signup == true)
 		puts "Probably not a vampire."
+	elsif match == false && willing == false && signup == false
+	  puts "Almost certainly a vampire"
 	elsif match == false && (willing == false || signup == false)
 		puts "Probably a vampire."
+	else
+	  puts "Results inconclusive"
 	end
 	obvious_name = true
 end
+end
+

```

Comment:
Lets break the evaluation conditions out of the questions.  It is always good to finish one job before starting another.  Collect all of the information you need about your employee and then move to eval separately.

Consider the optimum condition when running your code.  What can we do to run only necessary processes?  We know that if the employee has Drac or Fang in their name then they are definitely a vampire.  If that is the case, do we need to run any of the other checks?

Consider:

```
if name_matches_vampire_params
  "Definitely a vampire"
else
  if #other checks
    "Something"
  elsif #more conditions...

  else
  end
end
```